portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -40,10 +40,27 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
     PreparedQuery results = datastore.prepare(query);
 
+    String limitString = request.getParameter("limit");
+
+    // Convert the input to an int.
+    int limit;
+    try {
+      limit = Integer.parseInt(limitString);
+    } catch (NumberFormatException e) {
+      System.err.println("Could not convert to int: " + limitString);
+      limit = -1;
+    }
+
     ArrayList<String> comments = new ArrayList<String>();
+    int i = 0;
     for (Entity entity : results.asIterable()) {
-      String text = (String) entity.getProperty("text");
-      comments.add(text);
+      if (i < limit || limit == -1) {
+        String text = (String) entity.getProperty("text");
+        comments.add(text);
+        i++;
+      } else {
+        break;
+      }
     }
```

Comment:
Use [`FetchOptions`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/FetchOptions.html) with [`asIterable(FetchOptions fetchOptions`](https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PreparedQuery.html#asIterable-com.google.appengine.api.datastore.FetchOptions-) instead. 

In the catch for the NumberFormatException, just don't set the limit on FetchOptions in that case. The `-1` trick seems a little hacky.

Reply:
I implemented the FetchOptions and added a condition in the js file to account for "all".